### ISSUE TITLE
In game inputs from non english layouts should be treated as their qwerty equivalent (like SDL2 scancodes)

### ISSUE THREAD

**Issue by ghost** [godotengine/godot#8039]
Linux Kubuntu 16.10 Godot 2.1.2 x64 official
Self descriptive. After switching layouts, in game inputs from keyboard are ignored. 

Expected behavior: processing inputs as if they are from english layout.

**Comment by volzhs** [comment]
duplicated issue with #7106 ?

**Comment by ArcFutahito** [comment]
Nope. Editor works pretty much as intended it's just that different keyboard languages are not being taken in consideration during input.

**Comment by Hinsbart** [comment]
Mhh can't reproduce using a german keyboard layout, all keys seem to trigger the correct input. (OpenSUSE Tumbleweed, both 2.1.2 and master)

**Comment by ArcFutahito** [comment]
russian and ukrainian layouts not working

**Comment by Hinsbart** [comment]
@ArcFutahito Well german also has some special chars, they work fine however.
Also I just set my layout to russian and it seems to work too:
![keyboard](https://cloud.githubusercontent.com/assets/8281916/23962120/622a282e-09ad-11e7-8fdf-f335724c4073.png)

I can even change layouts while the editor is running...

**Comment by ArcFutahito** [comment]
I am talking game input, not editor input. Add german char to input map and watch editor fail.

**Comment by Hinsbart** [comment]
Alright.
Now it might be a duplicate of #2915? :P

**Comment by ArcFutahito** [comment]
Nope. Issue asks for support of non ascii binding, I ask for input to be layout independent.

**Comment by akien-mga** [comment]
>>  Now it might be a duplicate of #2915? :P

> Nope. Issue asks for support of non ascii binding, I ask for input to be layout independent.

So, yes, it's a duplicate, but you want a different solution. Doesn't mean it's not the same underlying issue :)

**Comment by ArcFutahito** [comment]
Well, not really? Binding a non ascii character is different from using non ascii characters with default english binds. Well, whatever makes your cookie crumble :D

**Comment by tomlechner** [comment]
In Linux, maybe this is addressed with the attached patch against master. 

I found this due to troubles with modifiers+non-letter keys on a dvorak keyboard causing dropped presses and releases, but maybe it applies also to non-english layouts? If not, I can open a different issue. This is my first time contributing, not totally sure the right place or way to do things!

In the code there seemed to be confusion between X11 keycodes (unique per physical key) and keysyms (might be many per physical key). Godot seems to expect a "scancode" that is basically unique per physical key, but is numerically similar to keysyms or'd with modifiers (unlike native x11 keycodes).

The old code used XLookupString to get a "keycode", but that actually returns a keysym filtered by modifiers, messing things up when doing "out of sync" modifier presses/releases for non-letter keys. Instead using XLookupKeysym(xevent,0) always returns the first in a list of keysyms for that keycode, which as far as I can tell seems to be unique, but I've only tested on a US keyboard (qwerty and dvorak).

[godot-patch-fix-x11keycode.txt](https://github.com/godotengine/godot/files/1906913/godot-patch-fix-x11keycode.txt)
